lib/msal-node/src/index.ts

```diff
@@ -118,4 +118,4 @@ export {
     AppTokenProviderResult,
 } from "@azure/msal-common";
```

Comment:
Do these changes need to be made for msal-common exports as well?

Reply:
`msal-common` is still `cjs`, once you change it to `"type": "module"` then the types will break and you'll have to move to the `.js` extension. At least thats as far as I understand

Reply:
This does also mean that if you remote `"type": "module"` from the distributed package.json, this would also solve the issue and with the smallest footprint. But I would consider this a hack more than a solution

Reply:
All of our libraries export both cjs and esm

Reply:
But they share a `package.json`, take a peek at the one both msal-common and msal-nodes ship.
This is the difference between them, and what broke typescript

Reply:
I'll be more specific. `msal-common` is still _written_ as a commonjs module for the simple fact that its type is not set to `"module"`-- So it defaults to `"commonjs"`. It and the rest are bundled for both `esm` and `cjs`, but the module type stays the same. Node decides which version to take wether you `import` or `require`, as outlined in your `"exports"` but can chew though both either way. the typescript type checker on the other hand, from what I understand, does not give 2 licks about how much effort we go though to bundle things universally, it knows either cjs _or_ esm. Depending on what the type of the module is-- which is decided by the package.json `"type"` field.

I'm sure you already know everything I said here, and I'm also sure I made mistakes while oversimplifying the issue, but thats the long version of why msal-common doesn't _need_ to do this yet, though it probably should